File: sources/org/neochess/core/Move.php
<?php

namespace org\neochess\core;

class Move
{
    private $fromSquare;
    private $toSquare;
    private $promotionPiece;
    
    public function __construct($fromSquare, $toSquare)
    {
        $this->fromSquare = $fromSquare;
        $this->toSquare = $toSquare;
    }

    public function getFromSquare()
    {
        return $this->fromSquare;
    }

    public function getToSquare()
    {
        return $this->toSquare;
    }

    public function setFromSquare($fromSquare)
    {
        $this->fromSquare = $fromSquare;
    }

    public function setToSquare($toSquare)
    {
        $this->toSquare = $toSquare;
    }
    
    public function getPromotionPiece()
    {
        return $this->promotionPiece;
    }

    public function setPromotionPiece($promotionPiece)
    {
        $this->promotionPiece = $promotionPiece;
    }
}

File: src/main/java/org/neochess/engine/Move.java

package org.neochess.engine;

public class Move {

    private int fromSquare;
    private int toSquare;
    private int promotionPiece;

    public Move(int fromSquare, int toSquare) {
        this.fromSquare = fromSquare;
        this.toSquare = toSquare;
    }

    public Move(int fromSquare, int toSquare, int promotionPiece) {
        this.fromSquare = fromSquare;
        this.toSquare = toSquare;
        this.promotionPiece = promotionPiece;
    }

    public int getFromSquare() {
        return fromSquare;
    }

    public void setFromSquare(int fromSquare) {
        this.fromSquare = fromSquare;
    }

    public int getToSquare() {
        return toSquare;
    }

    public void setToSquare(int toSquare) {
        this.toSquare = toSquare;
    }

    public int getPromotionPiece() {
        return promotionPiece;
    }

    public void setPromotionPiece(int promotionPiece) {
        this.promotionPiece = promotionPiece;
    }
}
